pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -181,6 +191,16 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
+	// Update the signer with the existing privateKey
+	r.signer, err = signer.Create(privateKey)
```

Comment:
I think the signer should be moved elsewhere, not sure if it'd make sense to have it in reconciler struct considering, this is not static anymore. Perhaps we can get the value at the call site.

Reply:
I dont know if it makes sense passing the privatekey all the way down to the nodeconfig

Reply:
Ideally I'd like to get the secret there in the nodeConfig instead of checking in the reconcile method. This way we don't have to pass the objects around. The downside to this approach is we cannot fail early in the reconcile. Previously we were assuming that signer will be invariant in all reconciles which is not the case anymore. So, to me it is better to fail at the location where we are acting up on it. However, this is a nit again. I don't see that much of a problem of with this approach. Just that it looks a bit different to what I would have implemented

Reply:
I think that approach is valid, however I think failing early is more important right now, im inclined to leave the code as it is.